book/src/semantics/functions.md

```diff
@@ -0,0 +1,68 @@
+# Functions
+
+## Function call
+
+The major distinction between PRQL and today's conventional programming
+languages such as C or Python is the function call syntax.
+It consists of the function name followed by arguments separated by whitespace.
+
+```prql_no_test
+function_name arg1 arg2 arg3
+```
+
+If one of the arguments is also a function call, it must be encased in parentehsis,
+so we know where arguments of inner function end and the arguments of outer function start.
+
+```prql_no_test
+outer_func arg_1 (inner_func arg_a, arg_b) arg_2
+```
+
+## Pipeline
+
+There is a alternative way of calling functions: using a pipeline.
+Regardless of whether the pipeline is delimited by pipe symbol `|` or a new line,
+the pipeline is equivalent to applying each of functions as the last argument of the next function.
+
+```prql_no_test
+a | foo 3 | bar 'hello' 'world' | baz
+```
+
+... is equivalent to ...
+
+```prql_no_test
+baz (bar 'hello' 'world' (foo 3 a))
+```
+
+You may not have noticed, but transforms are regular functions too!
+
+```elm
```

Comment:
(fyi, for the book, we can write `prql` or `prql_no_test`; it's only GH which requires `elm`. But no need to change)